### DISCUSSION TITLE
Best way to make automatic git version and autogenerate changelog / revision history page

### DISCUSSION THREAD

**Discussion by TheLeftMoose** [squidfunk/mkdocs-material#7275] [General]
Hi great people

I am looking for advise to how to automate git version + changelog with a simple Trunk Based Development pattern.

I get mike can do fancy stuf in the UI, but firstly I want to be in control of the release mechanism it self.

Any help is appreciated. Thanks in advance.

**Comment by HonkingGoose** [comment]
I think the [Semantic Release](https://github.com/semantic-release/semantic-release) bot is what you're looking for.

The basic idea is:

1. You follow the [SemVer 2.0 specifications](https://semver.org/) for all commits on your release branch, let's assume that's `main`
2. Semantic Release bot decides based on the commit message on `main` what kind of release to make: `patch`, `minor`, `major`.
3. Semantic Release bot cuts the release and uses the commit messages as changelog items.
4. Done.

Configuring Semantic Release can be tricky, so for that please read the docs. 🙂
